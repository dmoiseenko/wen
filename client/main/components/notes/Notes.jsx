import React from 'react';
import PropTypes from 'prop-types';

import Note from './Note';


export default function Notes({ notes, loading }) {
  return (
    <div className="content">
      <ul>
        {
          !loading &&
          notes.map(note =>
            (
              <Note
                key={note.id}
                note={note}
              />
            ))
        }
      </ul>
    </div>
  );
}

Notes.propTypes = {
  notes: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};

Notes.defaultProps = {
  notes: [],
};
