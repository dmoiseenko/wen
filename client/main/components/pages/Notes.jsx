import React from 'react';

import NotesContainer from '../notes/Notes.container';
import NoteFormContainer from '../notes/NoteForm.container';


export default () => (
  <div className="section">
    <div className="container">
      <h1 className="title is-2">
        Notes
      </h1>
      <NotesContainer />
      <NoteFormContainer />
    </div>
  </div>
);
