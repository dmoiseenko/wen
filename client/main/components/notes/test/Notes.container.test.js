import * as container from '../Notes.container';


it('propsMapper should map props correctly', () => {
  const data = {
    notes: 'notes',
    loading: 'loading',
    subscribeToMore: 'subscribeToMore'
  };

  const actual = container.propsMapper({ data });
  expect(actual).toMatchSnapshot();
});
