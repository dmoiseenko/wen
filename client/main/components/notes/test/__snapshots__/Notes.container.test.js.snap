// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propsMapper should map props correctly 1`] = `
Object {
  "loading": "loading",
  "notes": "notes",
}
`;
